src/curve/pairing.rs

```diff
@@ -0,0 +1,76 @@
+//! Pairing operations for the Pluto curve.
+
+use super::*;
+
+// TODO: I believe we will definitely need to construct GF(17) outright, as we will need to perform
+// arithmetic operations on it.
+
+// TODO: I think we also need to create something like a `struct Torsion<const R: usize>` to have a
+// representation of the r-torsion points on the curve. These may have to just be for our curve and
+// not generic as making them generic could be quite hard.
+
+/// Compute the Tate pairing of two points on the curve.
+pub fn pairing<const R: usize>(
+  p: AffinePoint<PlutoExtendedCurve>,
```

Comment:
should the input be from two different curve groups (asymmetric paring). If we look at the one in kzg we take a point in the F_q (the base field) and then a point in F_{q^2} (the extension field)

Reply:
The groups consist of points on the curve though :)

Reply:
Oh True! Should we restrict them at all? So that we know they are from the right groups.